### DISCUSSION TITLE
Building a near Real Time Model in Power BI

### DISCUSSION THREAD

**Discussion by mlonsk** [TabularEditor/TabularEditor3#1196] [Resources]
@bernatagulloesbrina has a new blog using some nice C# scripts that creates and updates partitions to always have the newest data in the semantic model.
https://www.esbrina-ba.com/building-my-first-near-real-time-model-in-power-bi/

**Comment by bernatagulloesbrina** [comment]
thank you for sharing! 
now if anyone has any hint on the best way to synchronize it with a ETL running on AWS I would appreciate it!

**Reply by otykier** [reply]
Hi @bernatagulloesbrina 
Thanks for writing the article. While I certainly enjoy seeing Tabular Editor used to solve issues in creative ways, I can't help feeling that using PowerShell to run Tabular Editor to execute a C# script that creates/refreshes partitions, is a bit of a "crossing the stream to get water" situation.

Did you know that you can execute TMSL commands directly from PowerShell, using the [`Invoke-ASCmd` cmdlet](https://learn.microsoft.com/en-us/powershell/module/sqlserver/invoke-ascmd?view=sqlserver-ps)? With TMSL scripts, you can create and refresh partitions (and more) - and by using the [sequence command](https://learn.microsoft.com/en-us/analysis-services/tmsl/sequence-command-tmsl?view=asallproducts-allversions) the whole thing is even executed within a transaction, which gets automatically rolled back in case something fails (so you don't leave your model with empty partitions, if the refresh fails for some reason, for example).

With regards to your question:

> now if anyone has any hint on the best way to synchronize it with a ETL running on AWS I would appreciate it!

I know absolutely nothing about ETL on AWS, but wouldn't it be possible to instruct the ETL process to call a web function once the ETL completes? Sort of like a webhook? In that case, you can set up your PowerShell script to run in an [Azure Runbook with a HTTP trigger](https://learn.microsoft.com/en-us/azure/automation/automation-webhooks?tabs=portal).

**Comment by bernatagulloesbrina** [comment]
Thank you Daniel!

I'm totally biased towards using tabular editor scripts, i know.
I did see that is possible to run tmsl commands from powershell, but
building the command is way beyond my knowledge of powershell

I'll check your link, but I'm afraid that it will force me to learn yet two
or three extra programming languages 😀

El dl., 11 de des. 2023, 11:41, Daniel Otykier ***@***.***>
va escriure:
